config/src/test/java/org/axonframework/config/SimpleStatefulCommandHandlingComponentTest.java

```diff
@@ -0,0 +1,323 @@
+            myModelId -> EventCriteria.forAnyEventType().withTags(new Tag("Course", myModelId)),
+            courseEventStateApplier,
+            Course::new,
+            DEFAULT_CONTEXT
+    );
+
+
+    ModelRegistry registry = SimpleModelRegistry
+            .create("MyModelRegistry")
+            .registerModel(
+                    String.class,
+                    Student.class,
+                    (id, context) -> studentRepository.loadOrCreate(id, context).thenApply(ManagedEntity::entity)
+            )
+            .registerModel(
+                    String.class,
+                    Course.class,
+                    (id, context) -> courseRepository.loadOrCreate(id, context).thenApply(ManagedEntity::entity)
+            );
+
+    /**
+     * Tests that the {@link StatefulCommandHandlingComponent} can handle a singular model command.
+     */
+    @Test
+    void canHandleSingularModelCommand() {
+        var component = StatefulCommandHandlingComponent
+                .create("MyStatefulCommandHandlingComponent", registry)
+                .subscribe(
+                        new QualifiedName(ChangeStudentNameCommand.class),
+                        (command, model, context) -> {
+                            ChangeStudentNameCommand payload = (ChangeStudentNameCommand) command.getPayload();
+                            Student student = model.getModel(Student.class, payload.id()).join();
+                            appendEvent(context, new StudentNameChangedEvent(student.id, payload.name()));
+                            return MessageStream.empty().cast();
+                        });
+
+        updateStudentName(component, "my-studentId-1", "name-1");
+        verifyStudentName("my-studentId-1", "name-1");
+        updateStudentName(component, "my-studentId-1", "name-2");
+        verifyStudentName("my-studentId-1", "name-2");
+        updateStudentName(component, "my-studentId-1", "name-3");
+        verifyStudentName("my-studentId-1", "name-3");
+        updateStudentName(component, "my-studentId-1", "name-4");
+        verifyStudentName("my-studentId-1", "name-4");
+
+        updateStudentName(component, "my-studentId-2", "name-5");
+        verifyStudentName("my-studentId-1", "name-4");
+        verifyStudentName("my-studentId-2", "name-5");
+    }
+
+    private void verifyStudentName(String id, String name) {
+        AsyncUnitOfWork uow = new AsyncUnitOfWork();
+        uow.executeWithResult(context ->
+                                      studentRepository
+                                              .load(id, context)
+                                              .thenAccept(student -> assertEquals(name, student.entity().name))
+        ).join();
+    }
+
+    /**
+     * Tests that the {@link StatefulCommandHandlingComponent} can handle a command that targets multiple models at the
+     * same time, in the same transaction.
+     */
+    @Test
+    void canHandleCommandThatTargetsMultipleModels() {
```

Comment:
```suggestion
    @Test
    void canHandleCommandThatTargetsMultipleModelsInTheSameTransaction() {
```

I think with proper test name (or you can use `@DisplayName`) the comment is obsolete.

Reply:
I thought about it more and I think `canHandleCommandThatTargetsMultipleModels` it's just enough, because transaction management it's not the responsibility of the handler.